cl/assets/static-global/js/webhooks-page.js

```diff
@@ -0,0 +1,120 @@
+    (target_trigger === 'webhook-filter-form' || target_trigger === 'webhooks-paginator')
+  ) {
+    //Show the loader spinner when sending the webhook filter or paginator request
+    $('#webhook_list_loader').removeClass('hidden');
+    $('#webhook_table_body').addClass('hidden');
+  }
+  if (e.detail.target.id === 'webhook_table_body' && target_trigger === 'webhooks-paginator' && !e.detail.isError) {
+    //Scroll toward the results before the pagination request.
+    document.getElementById('webhook-filter-form').scrollIntoView();
+  }
+});
+
+htmx.on('htmx:afterSwap', (e) => {
+  //Load the form over the modal
+  let webhook_form = document.getElementById('webhooks-body');
+  if (e.detail.target.id === 'webhooks-body') {
+    // If the user already have a webhook configured for each type of event, show a message.
+    let event_type_options = document.getElementById('id_event_type').options.length;
+    if (event_type_options === 0) {
+      webhook_form.innerHTML =
+        "<b class='text-center'>You already have a webhook configured for each type of event. Please delete one before making another.</b>";
+    }
+    //Toggle form modal
+    $('#webhook-modal').modal('toggle');
+  }
+
+  if (e.detail.target.id === 'webhooks-body-testing') {
+    //Toggle the webhook testing modal.
+    $('#webhook-modal').modal('toggle');
+  }
+
+  if (e.detail.target.id === 'webhook-test-sent') {
+    // Show the View in Logs button after send the test webhook request.
+    $('#webhook-test-sent').removeClass('hidden');
+  }
+
+  let target_trigger = e.detail.requestConfig.headers['HX-Trigger'];
+  if (
+    e.detail.target.id === 'webhook_table_body' &&
+    (target_trigger === 'webhook-filter-form' || target_trigger === 'webhooks-paginator') &&
+    !e.detail.isError
+  ) {
+    //Hide the loader spinner and show the table after loading the filter results.
+    $('#webhook_list_loader').addClass('hidden');
+    $('#webhook_table_body').removeClass('hidden');
+  }
+});
+
+htmx.on('htmx:beforeSwap', (e) => {
+  /*
+  After submitting the POST request, toggle the modal back to hidden.
+  */
+  if (e.detail.target.id === 'webhook-form' && !e.detail.xhr.response) {
+    $('#webhook-modal').modal('toggle');
+
+    $('#webhook_table_body').addClass('hidden');
+    $('#webhook_list_loader').removeClass('hidden');
+
+    e.detail.shouldSwap = false;
+  }
+
+  let webhook_title = document.querySelector('#webhook-modal .modal-title');
+  let webhook_form = document.getElementById('webhooks-body');
+  let webhook_form_testing = document.getElementById('webhooks-body-testing');
+  let target_trigger = e.detail.requestConfig.headers['HX-Trigger'];
+  if (e.detail.target.id === 'webhooks-body') {
+    // Clean and set the right title according to the HTMX request.
+    webhook_form.innerHTML = '';
+    webhook_form_testing.innerHTML = '';
+    if (target_trigger === 'add-webhook') {
+      webhook_title.innerHTML = 'Add a webhook endpoint';
```

Comment:
I can't explain why this is better English, but I guess when it's in a title you skip words:

```suggestion
      webhook_title.innerHTML = 'Add webhook endpoint';
```

Reply:
Changed.